### PR TITLE
Change `ValidationSummaryTagHelper`'s property to `ValidationSummary` enum

### PR DESCRIPTION
**Pull request by dougbu**
- #1685
- update tests and samples to match
- remove tests for case-insensitivity and invalid property values

Note .cshtml author is responsible for adding `@using Microsoft.AspNet.Mvc.TagHelpers`
- Microsoft.AspNet.Mvc.TagHelpers has no startup code and `MvcRazorHost` does
  not read `MvcOptions` anyhow

### CODE REVIEW

**Comment by dougbu** [comment]
/cc @DamianEdwards @NTaylorMullen @pranavkm

regarding the question I asked in #1685 
> what behaviour would you like for undefined ValidationSummary values e.g. asp-validation-summary="ValidationSummary)23"? compiler would be happy but should the tag helper throw or ignore the value?

I chose to ignore the value.  Only `ValidationSummary.ModelOnly` and `ValidationSummary.All` do anything.

File: test/Microsoft.AspNet.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs [hunk 1]
```diff
@@ -93,7 +93,7 @@ public class ValidationSummaryTagHelperTest
             // Arrange
             var validationSummaryTagHelper = new ValidationSummaryTagHelper
             {
-                ValidationSummaryValue = "ModelOnly"
+                ValidationSummary = ValidationSummary.All,
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs]: this was a small nit: noticed tests did not confirm `excludePropertyErrors` argument to `GenerateValidationSummary()` was `false` in this case

**Comment by dougbu** [comment]
got a couple of answers from @DamianEdwards.  closing this to make it obvious the current iteration isn't ready.

**Comment by dougbu** [comment]
will update the branch shortly...

**Comment by dougbu** [comment]
@NTaylorMullen @pranavkm updated and ready to go

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by dougbu** [comment]
Closed via 9b2a9e397649a928f85f323fe44a915af260035a

**Pull request closed without merging**